File: java/parser/src/main/java/gov/lbl/parser/parser/Output_expression_listVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Expression;
import gov.lbl.parser.domain.Output_expression_list;

import static java.util.stream.Collectors.toList;

import java.util.List;

public class Output_expression_listVisitor  extends modelicaBaseVisitor<Output_expression_list> {
    @Override
    public Output_expression_list visitOutput_expression_list(modelicaParser.Output_expression_listContext ctx) {
        ExpressionVisitor expressionVisitor = new ExpressionVisitor();
        List<Expression> output_expressions = ctx.expression() == null ? null : ctx.expression()
                                                                                    .stream()
                                                                                    .map(expression -> expression.accept(expressionVisitor))
                                                                                    .collect(toList());
        return new Output_expression_list(output_expressions);
    }
}

File: jsParser/parser/Output_expression_listVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor;
const Output_expression_list = require('../domain/Output_expression_list');

const ExpressionVisitor = require('./ExpressionVisitor');

class Output_expression_listVisitor {
    constructor() {
        modelicaVisitor.call(this);
        return this;
    }
    visitOutput_expression_list(ctx) {
        var output_expressions = [];

        if (ctx.expression()) {
            var expressionVisitor = new ExpressionVisitor.ExpressionVisitor();
            ctx.expression().forEach(expr => {
                output_expressions.push(expressionVisitor.visitExpression(expr));
            });
        }
        return new Output_expression_list.Output_expression_list(output_expressions);
    }
};

Output_expression_listVisitor.prototype = Object.create(modelicaVisitor.prototype);

exports.visitOutput_expression_list = this.visitOutput_expression_list;
exports.Output_expression_listVisitor = Output_expression_listVisitor;